glacier_mapping/train.py

```diff
@@ -136,13 +145,30 @@ def log_images(writer, frame, batch, epoch, stage="train"):
     Return:
         Images Logged onto tensorboard
     """
-    pm = lambda x: x.permute(0, 3, 2, 1)
+    pm = lambda x: x.permute(0, 3, 1, 2)
     squash = lambda x: (x - x.min()) / (x.max() - x.min())
 
     x, y = batch
-    y_hat = torch.sigmoid(frame.infer(x))
+    if frame.multi_class:
+        y_hat = torch.nn.Softmax(3)(frame.infer(x))
+    else:
+        y_hat = torch.sigmoid(frame.infer(x))
+    y = torch.flatten(y.permute(0, 1, 3, 2), start_dim=2)
+    y_hat = torch.flatten(y_hat.permute(0, 1, 3, 2), start_dim=2)
     if epoch == 0:
-        writer.add_image(f"{stage}/x", make_grid(pm(squash(x))), epoch)
-        writer.add_image(f"{stage}/y", make_grid(pm(y)), epoch)
+        writer.add_image(f"{stage}/x", make_grid(pm(squash(x[:, :, :, :3]))), epoch)
+        writer.add_image(f"{stage}/y", make_grid(y.unsqueeze(1)), epoch)
 
-    writer.add_image(f"{stage}/y_hat", make_grid(pm(y_hat)), epoch)
+    writer.add_image(f"{stage}/y_hat", make_grid(y_hat.unsqueeze(1)), epoch)
+
+def update_metrics(main_metrics, batch_metrics):
```

Comment:
What does this code do?

Reply:
It logs all predicted classes maps. If we have n classes, it logs n maps. It was logging one channel only before

Reply:
Sorry, I didn't mean this particular line, but block that you inserted starting from around here. Did we have an update_metrics function before? Or is this new / what is its motivation?

Reply:
We didn't. We updated metrics directly before by making a dataframe out of a list of dict and take the mean, as we had one value per metric. Since we have multiple classes now, update takes care of updating an array.

Reply:
Hm, okay. It makes sense to create a new function, and it's nice how concise it is. I wish it were a little more obvious what it did, but I can't think of a better way to do it right now.

Reply:
I commented it to make it clearer